tests/test-extensions/petstore-extension/src/main/java/org/mule/test/petstore/extension/PetAdoptionLimitingSource.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.test.petstore.extension;
+
+import static org.mule.runtime.extension.api.annotation.param.MediaType.TEXT_PLAIN;
+import org.mule.runtime.api.exception.MuleException;
+import org.mule.runtime.extension.api.annotation.metadata.MetadataScope;
+import org.mule.runtime.extension.api.annotation.param.MediaType;
+import org.mule.runtime.extension.api.annotation.param.Parameter;
+import org.mule.runtime.extension.api.runtime.operation.Result;
+import org.mule.runtime.extension.api.runtime.source.PollContext;
+import org.mule.runtime.extension.api.runtime.source.PollingSource;
+import org.mule.runtime.extension.api.runtime.source.SourceCallbackContext;
+
+import java.util.List;
+
+@MetadataScope(outputResolver = PollingSourceMetadataResolver.class)
+@MediaType(TEXT_PLAIN)
+public class PetAdoptionLimitingSource extends PollingSource<String, Integer> {
```

Comment:
why a new source for this? It'd be great to test it with an existing one since the real use case is injecting this functionality into existing sources

Reply:
I created a new source because I wanted to know in which poll was each item polled, and I also wanted to have an easy way to configure the watermark of each item. 

I can take some other tests and parameterize them so that they run with other sources using this parameter with a value of 1 or 2 for example, would that be ok?